public/user.php

```diff
@@ -28,332 +28,353 @@
+
+        $status = 'Scheduled';
+        $lastrun = 'NA';
+
+        $lastjobid = $ClientJobSchedule->GetLastJobId();
+        if ($lastjobid) {
+            $ClientJob = new ClientJob();
+            $ClientJob->Id = $lastjobid;
+            switch ($ClientJob->GetStatus()) {
+                case CDASH_JOB_RUNNING:
+                    $status = 'Running';
+                    $ClientSite = new ClientSite();
+                    $ClientSite->Id = $ClientJob->GetSite();
+                    $status .= ' (' . $ClientSite->GetName() . ')';
+                    $lastrun = $ClientJob->GetStartDate();
+                    break;
+                case CDASH_JOB_FINISHED:
+                    $status = 'Finished';
+                    $ClientSite = new ClientSite();
+                    $ClientSite->Id = $ClientJob->GetSite();
+                    $status .= ' (' . $ClientSite->GetName() . ')';
+                    $lastrun = $ClientJob->GetEndDate();
+                    break;
+                case CDASH_JOB_FAILED:
+                    $status = 'Failed';
+                    $ClientSite = new ClientSite();
+                    $ClientSite->Id = $ClientJob->GetSite();
+                    $status .= ' (' . $ClientSite->GetName() . ')';
+                    $lastrun = $ClientJob->GetEndDate();
+                    break;
+                case CDASH_JOB_ABORTED:
+                    $status = 'Aborted';
+                    $lastrun = $ClientJob->GetEndDate();
+                    break;
             }
-
-            $xml .= '<jobschedule>';
-            $xml .= add_XML_value('id', $scheduleid);
-            $xml .= add_XML_value('projectid', $Project->Id);
-            $xml .= add_XML_value('projectname', $Project->GetName());
-            $xml .= add_XML_value('status', $status);
-            $xml .= add_XML_value('lastrun', $lastrun);
-            $xml .= add_XML_value('description', $ClientJobSchedule->GetDescription());
-            $xml .= '</jobschedule>';
         }
+
+        $xml .= '<jobschedule>';
+        $xml .= add_XML_value('id', $scheduleid);
+        $xml .= add_XML_value('projectid', $Project->Id);
+        $xml .= add_XML_value('projectname', $Project->GetName());
+        $xml .= add_XML_value('status', $status);
+        $xml .= add_XML_value('lastrun', $lastrun);
+        $xml .= add_XML_value('description', $ClientJobSchedule->GetDescription());
+        $xml .= '</jobschedule>';
     }
+}
 
-    // Go through the public projects
-    $project = pdo_query("SELECT name,id FROM project WHERE id
-                        NOT IN (SELECT projectid as id FROM user2project
-                        WHERE userid='$userid') AND public='1' ORDER BY name");
-    $j = 0;
-    if ($CDASH_USE_LOCAL_DIRECTORY == '1') {
-        if (file_exists('local/user.php')) {
-            include_once 'local/user.php';
-        }
+// Find all the public projects that this user is not subscribed to.
+$pdo = get_link_identifier()->getPdo();
```

Comment:
Merely for the sake of consistency and readability do you think we should uppercase this variable name and define it near the top of the file? I wonder this because it's declared `global` it in the `ReportLastBuild` function below.